RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronSeedProducer.cc

```diff
@@ -0,0 +1,185 @@
+// -*- C++ -*-
+//
+// Package:    ElectronProducers
+// Class:      LowPtGsfElectronSeedProducer
+//
+/**\class LowPtGsfElectronSeedProducer RecoEgamma/ElectronProducers/src/LowPtGsfElectronSeedProducer.cc
+ Description: EDProducer of ElectronSeed objects
+ Implementation:
+     <Notes on implementation>
+*/
+// Original Author:  Robert Bainbridge
+
+#include "LowPtGsfElectronSeedProducer.h"
+#include "DataFormats/EgammaReco/interface/ElectronSeed.h"
+#include "DataFormats/EgammaReco/interface/ElectronSeedFwd.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "DataFormats/ParticleFlowReco/interface/PreIdFwd.h"
+#include "DataFormats/ParticleFlowReco/interface/PFCluster.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackBase.h"
+#include "FastSimulation/BaseParticlePropagator/interface/BaseParticlePropagator.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+#include "TMath.h"
+
+LowPtGsfElectronSeedProducer::LowPtGsfElectronSeedProducer( const edm::ParameterSet& iConfig ) :
+  tracks_()
+{
+  tracks_ = consumes<reco::TrackCollection>( iConfig.getParameter<edm::InputTag>("tracks") );
+  clusters_ = consumes<reco::PFClusterCollection>( iConfig.getParameter<edm::InputTag>("clusters") );
+  produces<reco::ElectronSeedCollection>();
+  produces<reco::PreIdCollection>();
+}
+
+LowPtGsfElectronSeedProducer::~LowPtGsfElectronSeedProducer() {}
+
+void LowPtGsfElectronSeedProducer::beginRun( edm::Run const&, 
+					     edm::EventSetup const& es ) 
+{
+  edm::ESHandle<MagneticField> field;
+  es.get<IdealMagneticFieldRecord>().get(field);
+  field_ = field->inTesla(GlobalPoint(0,0,0));
+}
```

Comment:
@mverzett In GoodSeedProducer::beginRun(), the following lines are added, but do not seem to do anything. See: https://github.com/CMSBParking/cmssw/blob/CMSSW_9_4_X/RecoParticleFlow/PFTracking/plugins/GoodSeedProducer.cc#L536-L537 Do we need them?...